tests/django_back_end/unit_of_work_test.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+import pytest
+from django.utils import timezone
+
+from testing.domain import factories
+from toy_settings.application import unit_of_work
+from toy_settings.django_back_end.unit_of_work import DjangoCommitter
+
+pytestmark = pytest.mark.django_db(transaction=True)
+
+
+def test_insert_indexes_out_of_order_raises_StaleState():
+    committer = DjangoCommitter()
+
+    # set a setting
+    committer.handle(
+        factories.Set(key="FOO", value="42", timestamp=timezone.now(), idx=0)
+    )
+
+    # two concurrent processes attempt to change the setting...
+    # ... the first succeeds
+    committer.handle(
+        factories.Changed(key="FOO", new_value="43", timestamp=timezone.now(), idx=1)
+    )
+
+    # ... the second fails
+    with pytest.raises(unit_of_work.StaleState):
+        committer.handle(
+            factories.Changed(
+                key="FOO", new_value="99", timestamp=timezone.now(), idx=1
+            )
+        )
```

Comment:
Nitpick: This test doesn't match my expectations based on the test name.

I would have expected to see events with indexes in an order more like this:
- 0
- 2
- 1

As it is, this test only proves failure on a duplicate index, not an out of order one.

Reply:
misnamed tests -- I can't work out how to test the out of order one because the trigger doesnt get added by Django

Reply:
I m wrong -- Django _does_ add the trigger, so I've updated the test to try both out of order and 'concurrent' inserts